src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/domain/goal/Goal.kt

```diff
@@ -0,0 +1,19 @@
+package uk.gov.justice.digital.hmpps.educationandworkplanapi.domain.goal
+
+import java.time.Instant
+import java.time.LocalDate
+import java.util.UUID
+
+data class Goal(
+  val reference: UUID,
+  val title: String,
+  val reviewDate: LocalDate,
+  val type: GoalType,
+  val steps: List<Step>,
+  val status: GoalStatus,
```

Comment:
Not for this PR, but I think this type of thing will come out as we work through the dev and establish our patterns in this respect, but I'm now back to wondering if the all args constructor should be private, and we have other public constructors that provide values for certain fields (like we played around with yesterday)
I'm specifically thinking of `status` ,,,, when would I want to create (from a business domain point of view) a `Goal` with a status other than `NOT_STARTED` ?

I guess this will come out when we start thinking about the domain service class, and who/where we expect a `Goal` to be created 🤔

Reply:
Perhaps we should default status to `NOT_STARTED`